data/list/basic.lean

```diff
@@ -795,6 +795,27 @@ theorem update_nth_eq_take_cons_drop (a : α) {n l} (h : n < length l) :
   update_nth l n a = take n l ++ a :: drop (n+1) l :=
 by rw [update_nth_eq_modify_nth, modify_nth_eq_take_cons_drop _ h]
 
+@[simp] lemma update_nth_eq_nil {β} (l : list β) (n : ℕ) (a : β) : l.update_nth n a = [] ↔ l = [] :=
+begin
+split,
+show list.update_nth l n a = list.nil → l = list.nil, 
+{
+  induction l,
+  case list.nil {
+    intros,
+    assumption
+  },
+  case list.cons {
+    induction n,
+    all_goals { contradiction }
+  }
+},
+-- show l = list.nil → list.update_nth l n a = list.nil,
```

Comment:
why is this `show` commented out?

Reply:
This show mysteriously fails. I've added a comment to this effect. I can't tell if it's a bug or I'm missing something.